code/game/machinery/crafting_station.dm

```diff
@@ -0,0 +1,575 @@
+	icon = 'icons/obj/machines/crafting_station.dmi'
+	icon_state = "craft"
+	circuit = /obj/item/electronics/circuitboard/crafting_station
+	density = TRUE
+	anchored = TRUE
+	layer = 2.8
+
+	var/power_cost = 250
+
+	var/working = FALSE
+	var/start_working
+	var/work_time = 20 SECONDS
+	var/mat_efficiency = 15
+	var/storage_capacity = 40
+	var/list/stored_material = list()
+	var/list/accepted_material = list (MATERIAL_PLASTEEL, MATERIAL_STEEL, MATERIAL_PLASTIC, MATERIAL_WOOD, MATERIAL_CARDBOARD)
+	var/list/needed_material_gunpart = list(MATERIAL_PLASTEEL = 5)
+	var/list/needed_material_armorpart = list(MATERIAL_STEEL = 20, MATERIAL_PLASTIC = 20, MATERIAL_WOOD = 20,MATERIAL_CARDBOARD = 20)
+	var/list/needed_material_ammo = list(MATERIAL_STEEL = 10, MATERIAL_CARDBOARD = 5)
+
+	// A vis_contents hack for materials loading animation.
+	var/tmp/obj/effect/flicker_overlay/image_load
+
+/obj/machinery/craftingstation/Initialize()
+	. = ..()
+
+	image_load = new(src)
+
+
+/obj/machinery/craftingstation/examine(user)
+	. = ..()
+	var/list/matter_count_need_ammo = list()
+	var/list/matter_count_need_gunpart = list()
+	var/list/matter_count_need_armorpart = list()
+
+	for(var/_material in needed_material_ammo)
+		matter_count_need_ammo += "[needed_material_ammo[_material]] [_material]"
+
+	for(var/_material in needed_material_gunpart)
+		matter_count_need_gunpart += "[needed_material_ammo[_material]] [_material]"
+
+	for(var/_material in needed_material_armorpart)
+		matter_count_need_armorpart += "[needed_material_ammo[_material]] [_material]"
+
+	var/list/matter_count = list()
+	for(var/_material in stored_material)
+		matter_count += " [stored_material[_material]] [_material]"
+
+	to_chat(user, SPAN_NOTICE("Materials required to craft ammunition: [english_list(matter_count_need_ammo)].\nMaterials required to craft gun parts: 5 [english_list(matter_count_need_gunpart)] \nMaterials required to craft armor parts: [english_list(matter_count_need_armorpart)] \nIt contains: [english_list(matter_count)]."))
+
+
+/obj/machinery/craftingstation/attackby(obj/item/I, mob/user)
+	if(default_part_replacement(I, user))
+		return
+
+	if(default_deconstruction(I, user))
+		return
+
+	if(istype(I, /obj/item/stack))
+		eat(user, I)
+		return
+	. = ..()
+
+/obj/machinery/craftingstation/attack_hand(mob/user as mob)
+	var/cog_stat = user.stats.getStat(STAT_COG)
+
+	var/list/options = list()
+	options[".20 Rifle ammunition"] = "srifle"
+	options[".25 Caseless Rifle ammunition"] = "clrifle"
+	options[".30 Rifle ammunition"] = "lrifle"
+	options[".35 Auto ammunition"] = "pistol"
+	options[".40 Magnum ammunition"] = "magnum"
+	options[".50 Shotgun Buckshot ammunition"] = "shot"
+	options[".50 Shotgun Beanbag ammunition"] = "bean"
+	options[".50 Shotgun Slug ammunition"] = "slug"
+	options[".60 Anti-Material ammunition"] = "antim"
+	options["Gun parts"] = "gunpart"
+	options["Armor parts"] = "armorpart"
```

Comment:
Can you make an associative list by just having 
```
	var/list/options = list(
	".20 Rifle ammunition" = "srifle",
	".25 Caseless Rifle ammunition" = "clrifle",
	".30 Rifle ammunition" = "lrifle",
	".35 Auto ammunition" = "pistol",
	".40 Magnum ammunition" = "magnum",
	".50 Shotgun Buckshot ammunition" = "shot",
	".50 Shotgun Beanbag ammunition" = "bean",
	".50 Shotgun Slug ammunition" = "slug",
	".60 Anti-Material ammunition" = "antim",
	"Gun parts" = "gunpart",
	"Armor parts"= "armorpart",
	)
```
Or something, this isn't your problem as many other associative lists in the code do this, but it bugs me.